### PR TITLE
mkgmap: 4565 → 4600

### PR DESCRIPTION
**Pull request by sikmir**
###### Motivation for this change
* [Changelog](http://www.mkgmap.org.uk/svn/commits)
* replace `fetchsvn` to `fetchurl`
* add `passthru.updateScript`

###### Things done
- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/mkgmap/default.nix [hunk 1]
```diff
@@ -9,34 +9,27 @@
...
+stdenv.mkDerivation rec {
   pname = "mkgmap";
-  inherit version;
+  version = "4600";
 
-  src = fetchsvn {
-    inherit sha256;
-    url = "https://svn.mkgmap.org.uk/mkgmap/mkgmap/trunk";
-    rev = version;
+  src = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/mkgmap/default.nix:19]: I think we should use fetchzip here.

File: pkgs/applications/misc/mkgmap/splitter/default.nix [hunk 2]
```diff
@@ -1,41 +1,37 @@
...
+stdenv.mkDerivation rec {
   pname = "splitter";
-  inherit version;
+  version = "597";
 
-  src = fetchsvn {
-    inherit sha256;
-    url = "https://svn.mkgmap.org.uk/mkgmap/splitter/trunk";
-    rev = version;
+  src = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/mkgmap/splitter/default.nix:18]: I think we should use fetchzip here.

File: pkgs/applications/misc/mkgmap/update.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl common-updater-scripts
+
+set -eu -o pipefail
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/mkgmap/update.sh:4]:
```suggestion
set -euo pipefail
```

File: pkgs/applications/misc/mkgmap/update.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+version=`curl -s $2 | sed -n 's#.*<span data-bind="text: current().version">\(.*\)</span>.*#\1#p'`
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/mkgmap/update.sh:6]:
```suggestion
version=$(curl -s $2 | sed -n 's#.*<span data-bind="text: current().version">\(.*\)</span>.*#\1#p')
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/mkgmap/default.nix (hunk 1 above)
Comment (sikmir) [reply pkgs/applications/misc/mkgmap/default.nix:19]: > The main difference between **fetchurl** and **fetchzip** is in how they store the contents. fetchurl will store the unaltered contents of the URL within the Nix store. fetchzip on the other hand will decompress the archive for you, making files and directories directly accessible in the future.

@SuperSandro2000 why? what is the main reason to choose fetchzip over fetchurl in general?

Comment (SuperSandro2000) [reply pkgs/applications/misc/mkgmap/default.nix:19]: with fetchzip you do not need to manually uncrompress the files.

Comment (sikmir) [reply pkgs/applications/misc/mkgmap/default.nix:19]: With fetchurl you do not need to manually uncompress as well.
What I see is:

> ... This is primarily useful for dynamically generated archives, such as GitHub's /archive URLs, where **the unpacked  content of the zip file doesn't change, but the zip file itself may** (e.g. due to minor changes in the compression algorithm, or changes in timestamps).

So fetchzip is kind of workaround for some cases, where fetchurl doesn't work properly due to mentioned above reasons.

**Pull request merged**